### PR TITLE
SNOW-1737253: Update docs for tz_convert/tz_localize variants to specify supported timezone formats

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1737253

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Update docs for tz_convert/tz_localize variants to specify supported timezone formats.

### CODE REVIEW

File: docs/source/modin/supported/dataframe_supported.rst [hunk 1]
```diff
@@ -481,10 +481,17 @@ Methods
...
+|                             |                                 |                                  | supported.                                         |
++-----------------------------+---------------------------------+----------------------------------+----------------------------------------------------+
+| ``tz_localize``             | P                               | ``axis``, ``level``, ``copy``    | ``N`` if timezone format is not supported.         |
+|                             |                                 | ``ambiguous``, ``nonexistent``   | Only timezones listed in ``pytz.all_timezones`` are|
+|                             |                                 |                                  | supported. So for example, ``UTC`` is supported but|
+|                             |                                 |                                  | ``UTC+/-<offset>`` such as ``UTC+09:00`` is not    |
+|                             |                                 |                                  | supported.                                         |
```
Comment (sfc-gh-jjiao) [inline docs/source/modin/supported/dataframe_supported.rst:494]: Nit:
"So" is unnecessary in "So for example", and we could also use the right punctuations to 
```
N if timezone format is not supported. Only timezones listed in pytz.all_timezones are supported. For example, UTC is supported, but UTC+/-<offset>, such as UTC+09:00 is not supported.
```

**Review by sfc-gh-jjiao — APPROVED** [review]
LGTM with a comment

File: CHANGELOG.md [hunk 2]
```diff
@@ -73,6 +73,7 @@
 - Removed unnecessary warning message about `convert_dtype` in `Series.apply`.
 - Improved generated SQL query for `head` and `iloc` when the row key is a slice.
 - Improved error message when passing an unknown timezone to `tz_convert` and `tz_localize` in `Series`, `DataFrame`, `Series.dt`, and `DatetimeIndex`.
+- Improved documentation for `tz_convert` and `tz_localize` in `Series`, `DataFrame`, `Series.dt`, and `DatetimeIndex` to specify the supported timezone formats.
```
Comment (sfc-gh-azhan) [inline CHANGELOG.md:76]: wrong version to add this message.

**Review by sfc-gh-azhan — CHANGES_REQUESTED** [review]

File: docs/source/modin/supported/dataframe_supported.rst (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply docs/source/modin/supported/dataframe_supported.rst]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (sfc-gh-helmeleegy) [reply CHANGELOG.md]: Fixed.

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**